slsim/lens.py

```diff
@@ -670,19 +698,25 @@ def source_light_model_lenstronomy(self, band=None):
             self._source_type == "extended"
             or self._source_type == "point_plus_extended"
         ):
-            if self.light_profile == "single_sersic":
-                source_models["source_light_model_list"] = ["SERSIC_ELLIPSE"]
+            source_models_list = []
+            kwargs_source_list = []
+            for source in self.source:
+                source_models_list.append(source.extended_source_light_model())
```

Comment:
for the different source redshifts, I think we need to save also the source_redshift_list as well, otherwise the danger is that people are using this setting and forget to also query the source redshift list